lib/internal/modules/esm/loader.js

```diff
@@ -157,12 +159,26 @@ function nextHookFactory(chain, meta, validate) {
       // Update only when hook is invoked to avoid fingering the wrong filePath
       meta.hookErrIdentifier = `${hookFilePath} '${hookName}'`;
 
-      validate(meta.hookErrIdentifier, ...args);
+      ReflectApply(
+        validate,
+        this,
+        ArrayPrototypeConcat(
+          meta.hookErrIdentifier,
+          args,
+        ),
+      );
```

Comment:
https://github.com/nodejs/node/blob/df5664440e447c2724003700b5a81da8fadc4f5a/doc/contributing/primordials.md#L272-L320
```suggestion
      validate(meta.hookErrIdentifier, args);
```

Reply:
Sorry, why not ReflectApply here? I thought you said not to change the signatures?

Reply:
I think that's an internal method, changing its signature should not be a problem, right?

Reply:
Yeah, it's internal and not a problem to do, but it's inconsistent (and more work).

Reply:
How would you do it without using spreading nor `ArrayPrototypeConcat`? I'm suggesting this because I believe that's the easiest, but if you think you can come up with a simpler solution, sure go ahead.

Reply:
> (ps if it shouldn't be used, why does it exist?)

I plan of introducing a linter rule that warns about this, I agree it's not reasonable to expect collaborators to remember all gotchas of primordials.